kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ccv2/Catalog.scala

```diff
@@ -0,0 +1,53 @@
+package io.delta.kernel.defaults.ccv2
+
+import io.delta.kernel.ccv2.ResolvedMetadata
+import io.delta.kernel.internal.actions.{Metadata, Protocol}
+import io.delta.kernel.utils.FileStatus
+
+////////////////////////////////
+// CatalogClient interactions //
+////////////////////////////////
+
+trait CatalogClient {
+  def resolveTable(tableName: String): ResolveTableResponse
+  def getCommits(): GetCommitsResponse
+}
+
+trait ResolveTableResponse {
+  def getPath: String
+  def getVersion: Long
+  def getProtocol: Option[Protocol]
+  def getMetadata: Option[Metadata]
+  def getSchemaString: Option[String]
+}
+
+trait GetCommitsResponse {
+  def getCommits(): Seq[FileStatus]
+}
+
+/////////////////////////////
+// CCv2Client interactions //
+/////////////////////////////
+
+trait CCv2Client {
+  def resolveTable(tableName: String): ResolvedMetadata
+}
+
+///////////////////////////
+// InMemoryCatalogClient //
+///////////////////////////
+
+class InMemoryCatalogClient extends CatalogClient {
+  override def resolveTable(tableName: String): ResolveTableResponse = ???
+  override def getCommits(): GetCommitsResponse = ???
+}
+
+////////////////////////
+// InMemoryCCv2Client //
+////////////////////////
+
+class InMemoryCCv2Client extends CCv2Client {
+  override def resolveTable(tableName: String): ResolvedMetadata = {
```

Comment:
```
// UC_REST_CLIENT
uc_catalog.delta.protocol.$field1

// GLUE_REST_CLIENT
glue_catalog.delta.protocol.$field1

Should the UC_CCV2_CLIENT and the GLUE_CCV2_CLIENT then translate to KERNEL?

kernel.delta.protocol
```

Q: if we give them a bag of properties, will they add a prefix? are they responsible for stripping the prefix when they give the bag of properties back to us?

Q: is the goal that the CCv2 clients should not have to do a bunch of translation / deep understanding?

**Intuition: Kernel should be able to commit a bag of properties and then get back that exact same bag of properties. Maybe the CCv2 client needs to add/strip some prefixes when it goes back and forth to the catalog.**

Reply:
the interface `Kernel <> CCv2Client` should be the bag of properties